src/atdf/peripheral.rs

```diff
@@ -9,13 +9,26 @@ pub fn parse_list(
 ) -> crate::Result<Vec<chip::Peripheral>> {
     let mut peripherals = vec![];
 
-    for module in el.children.iter() {
-        module.check_name("module")?;
+    for module in el.children.iter().filter_map(|node| {
+        let module = node.as_element().filter(|e| e.name == "module");
+
+        if module.is_none() {
+            log::warn!("Unhandled module element: {module:?}");
+        }
```

Comment:
I see you replaced the error with a warning here.  Was there a specific reason for this?  Like an ATDF having such elements and `atdf2svd` failing because of them?

Reply:
There aren't really any more unhandled elements, but this also handles CData elements, processing instructions and such.